### PR TITLE
Migrate integreatly-test-cases

### PR DESCRIPTION
**Pull request by b1zzu**
# Description

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -12,4 +12,5 @@
 ## Checklist
 - [ ] This change requires a documentation update <!-- Update JIRA with Affects -> Documentation -->
 - [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] I have added a test case that will be used to verify my changes
```
Comment (b1zzu) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: @davidffrench do you think will this line be enugh?

File: test-cases/OWNERS [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
+
+approvers:
+- mikenairn
+- pmccarthy
+- maleck13
+- philbrookes
+- grdryn
+- david-martin
```
Comment (damienomurchu) [inline test-cases/OWNERS:9]: Think this is probably duplication as any [root-level OWNERS](https://github.com/integr8ly/integreatly-operator/blob/master/OWNERS) will be able to approve PRs in the test-cases repo also

File: test-cases/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# Integreatly Test Cases
+
+- Test cases are located in the directory `/tests` and are organized in subdirectories by categories.
```
Comment (damienomurchu) [inline test-cases/README.md:3]:
```suggestion
 - Test cases are located in the directory `/test-cases` and are organized in subdirectories by categories. 
```

Comment (damienomurchu) [reply test-cases/README.md:3]: Need to update any other references also, fyi

File: test-cases/README.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+- When estimating the test do not count the time of reporting bugs, or debugging issues
+- Try to estimate the test as someone that is doing it for the first time
+
+## How to automate a test case and link it back
+
+1. Create the automated test in the appropriate test suite depending on the type and complexity of the test.
+
+   **Test suites:**
+
+   - [Functional Test Suite](https://github.com/integr8ly/integreatly-operator/tree/master/test)
```
Comment (damienomurchu) [inline test-cases/README.md:65]:
```suggestion
  - [Functional Test Suite](https://github.com/integr8ly/integreatly-operator/tree/master/test/functional)
```

**Comment by damienomurchu** [comment]
Looks good, few comments left

Re Prow, can we get an exception added to e2e tests aren't required to pass on a PR that only modifies `test-cases/*`?

File: test-cases/OWNERS (hunk 2 above)
Comment (b1zzu) [reply test-cases/OWNERS:9]: Yes, I thought so, but I saw that they are also in tests/OWNERS so maybe if the owners are not inheriting.. I don't know.. should I remove them?

File: test-cases/README.md (hunk 3 above)
Comment (b1zzu) [reply test-cases/README.md:3]: The README is located in the `/test-cases` directory but the tests (.md files) are in the `test-cases/tests` directory, so just `tests/`, but I should remove the `/`

**Comment by b1zzu** [comment]
> Re Prow, can we get an exception added to e2e tests aren't required to pass on a PR that only modifies test-cases/*?

We will try to run like this for bit, and if we see that many PR are blocked because of the e2e we will try to exclude them

File: openshift-ci/Dockerfile.tools [hunk 5]
```diff
@@ -17,4 +17,14 @@ RUN mkdir -p $GOPATH/src/github.com/operator-framework \
...
+
+# install nodejs
+ENV NODE_VERSION=12.16.3 \
+    NPM_CONFIG_CACHE=/tmp/.npm
+
+RUN curl -fsSLO --compressed "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.xz" \
+    && curl -fsSLO --compressed "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
+    && tar -xJf "node-v$NODE_VERSION-linux-x64.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
+    && rm "node-v$NODE_VERSION-linux-x64.tar.xz" \
+    && ln -s /usr/local/bin/node /usr/local/bin/nodejs
```
Comment (mikenairn) [inline openshift-ci/Dockerfile.tools:30]: This is going to conflict with the changes made here https://github.com/integr8ly/integreatly-operator/pull/725/files#diff-48abca7cd218705cd020054b38fa29cd. Those should be merged back soon, will need to update this after it merges before we can merge this.

File: openshift-ci/test/run [hunk 6]
```diff
@@ -10,11 +10,13 @@ WORK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 docker run --rm --entrypoint=/bin/sh ${IMAGE_NAME} -c 'operator-sdk version'
 docker run --rm --entrypoint=/bin/sh ${IMAGE_NAME} -c 'go version'
 docker run --rm --entrypoint=/bin/sh ${IMAGE_NAME} -c 'jq --version'
+docker run --rm --entrypoint=/bin/sh ${IMAGE_NAME} -c 'node --version'
 
 docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make vendor/check"
 docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make code/check"
 docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make code/compile"
 docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make test/unit"
+docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make test-cases/lint"
```
Comment (mikenairn) [inline openshift-ci/test/run:19]: Verified locally, works as expected, we will however have to wait and check it again once that other PR is merged before we can update openshift CI.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (davidffrench) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: @b1zzu Yes, and more importantly that it is enforced by the repo approvers on a PR

File: test-cases/OWNERS (hunk 2 above)
Comment (davidffrench) [reply test-cases/OWNERS:9]: @damienomurchu is correct. I am a root-level OWNER but not an OWNER of the test folder https://github.com/integr8ly/integreatly-operator/blob/master/test/OWNERS but I have rights to merge PRs there.

Comment (damienomurchu) [reply test-cases/OWNERS:9]: @b1zzu yes, should be fine to remove

**Comment by mikenairn** [comment]
/hold

Update and approve after https://github.com/integr8ly/integreatly-operator/pull/725

**Comment by mikenairn** [comment]
/hold cancel

The Dockerfile conflicts will need resolved before it can merge.

File: openshift-ci/Dockerfile.tools [hunk 7]
```diff
@@ -20,3 +20,12 @@ RUN wget -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux6
...
     && chmod +x /usr/local/bin/yq
+
+# install nodejs
+ENV NODE_VERSION=12.16.3 \
+    NPM_CONFIG_CACHE=/tmp/.npm
+
+RUN curl -fsSLO --compressed "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.xz" \
+    && tar -xJf "node-v$NODE_VERSION-linux-x64.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
+    && rm "node-v$NODE_VERSION-linux-x64.tar.xz" \
+    && ln -s /usr/local/bin/node /usr/local/bin/nodejs
```
Comment (mikenairn) [inline openshift-ci/Dockerfile.tools:31]: Works as expected :+1:

**Comment by mikenairn** [comment]
@b1zzu Did you add a lint test to CI for this?

/lgtm
/approve

**Comment by b1zzu** [comment]
@mikenairn yer here is the PR for the lint pipeline: https://github.com/openshift/release/pull/9114

The cmd is `make test-cases/lint`

**Comment by b1zzu** [comment]
@mikenairn can we skip the e2e tests and merge it anyway?

**Comment by mikenairn** [comment]
@b1zzu I guess we could for this one since it includes no code changes. I have no idea why the e2e tests are so up and down though.

**Comment by mikenairn** [comment]
I guess it makes sense, quay.io is down and some of our CSVs are pullingr images from there.

**Comment by b1zzu** [comment]
@mikenairn yes quay.io is still down, and this PR doesn't have any code changes, if we could merge it it would be great, there are other people waiting for this feature to be merged

**Pull request merged**